src/Microsoft.Health.Dicom.Blob.UnitTests/Features/ExternalStore/MockRequest.cs

```diff
@@ -0,0 +1,45 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Azure.Core;
+
+namespace Microsoft.Health.Dicom.Blob.UnitTests.Features.ExternalStore;
+
+internal class MockRequest : Request
```

Comment:
Something about `Request` couldn't be mocked via `NSubstitute`? Or this was easier?

Reply:
Yeah, `Headers` is not virtual or abstract so cannot be mocked https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Request.cs#L108